include/circt/Dialect/DC/DCCanonicalization.td

```diff
@@ -16,5 +16,13 @@
 include "circt/Dialect/DC/DC.td"
 include "mlir/IR/PatternBase.td"
 
+def HasOneOperand : Constraint<CPred<"$_self.size() == 1">, "has one operand">;
+def HasOneResult : Constraint<CPred<"$_self.size() == 1">, "has one result">;
+
+def EliminateSimpleJoinPattern : Pat<(JoinOp $arg), (replaceWithValue $arg),
```

Comment:
These could be folds, correct? There are a bunch of advantages to using folds rather than canonicalization patterns, which I'm pretty sure are the only way these can be used.

Reply:
Bear in mind that we're gonna want to do constant propagation analysis which uses folds. Dunno if that applies here -- just a general comment.